### PR TITLE
Fix a false negative for `Performance/ReverseEach` when safe navigation is between `reverse` and `each`

### PR DESCRIPTION
**Pull request by fatkodima**
Fixes https://github.com/rubocop/rubocop-performance/issues/390.

### CODE REVIEW

File: lib/rubocop/cop/performance/reverse_each.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ class ReverseEach < Base
         RESTRICT_ON_SEND = %i[each].freeze
 
         def_node_matcher :reverse_each?, <<~MATCHER
-          (send (call _ :reverse) :each)
+          ({send csend} (call _ :reverse) :each)
```
Comment (koic) [inline lib/rubocop/cop/performance/reverse_each.rb:30]:
```suggestion
          (call (call _ :reverse) :each)
```

**Comment by koic** [comment]
CI has been failed. Can you rebase with the latest master branch?

**Comment by fatkodima** [comment]
Updated.

File: spec/rubocop/cop/performance/reverse_each_spec.rb [hunk 2]
```diff
@@ -12,6 +12,8 @@
     expect_offense(<<~RUBY)
       array&.reverse.each { |e| puts e }
              ^^^^^^^^^^^^ Use `reverse_each` instead of `reverse.each`.
+      array&.reverse&.each { |e| puts e }
+             ^^^^^^^^^^^^^ Use `reverse_each` instead of `reverse.each`.
```
Comment (koic) [inline spec/rubocop/cop/performance/reverse_each_spec.rb:16]: Please separate this, an independent test instead of mixing it in the existing `it` block.

Comment (fatkodima) [reply spec/rubocop/cop/performance/reverse_each_spec.rb:21]: Hate inventing new names for tests 😅 Updated

**Pull request merged**